hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/TransactionProcessor.java

```diff
@@ -149,8 +153,18 @@ private HederaEvmTransactionResult safeCommit(
         return result;
     }
 
-    private record InvolvedParties(
-            @NonNull HederaEvmAccount sender, @Nullable HederaEvmAccount relayer, @NonNull Address receiverAddress) {}
+    private HederaEvmTransactionResult commitResourceExhaustion(
+            @NonNull final HederaEvmTransaction transaction,
+            @NonNull final HederaWorldUpdater updater,
+            @NonNull final HederaEvmContext context,
+            @NonNull final ResponseCodeEnum reason,
+            @NonNull final Configuration config) {
+        // Note these calls cannot fail, or processTransaction() above would have aborted right away
```

Comment:
(Example of a good comment needed because it is not true that "good code documents itself" (always). This is good code but there are semantics that are inexpressible in most programming languages.)